src/tsickle.ts

```diff
@@ -759,43 +797,87 @@ class Annotator extends ClosureRewriter {
       //
       // Both of these forms create a local name "a", which after
       // TypeScript CommonJS compilation will become some renamed
-      // variable like "module_1.a".  But a user might still use plain
-      // "a" in some JSDoc comment, so gather up these local names for
-      // imports and make an alias for each for JSDoc purposes.
+      // variable like "module_1.default" or "module_1.a" (for default vs named bindings,
+      // respectively).
+      // tsickle references types in JSDoc. Because the module prefixes are not predictable, and
+      // because TypeScript might remove imports entirely if they are only for types, the code below
+      // inserts an artificial `const prefix = goog.require` call for the module, and then registers
+      // all symbols from this import to be prefixed.
+      // Subtlety warning: for a default import, the import name is used as the prefix. This means
+      // references to the import do not have to be prefixed, as it already has the correct name.
       if (!this.options.untyped) {
-        let localNames: string[];
+        let symbols: ts.Symbol[] = [];
+        const typeChecker = this.program.getTypeChecker();
         if (importClause.name) {
-          // import a from ...;
```

Comment:
Can you leave in these comments with the example syntax of which sort of import you're handling?  I found them helpful for understanding the nest of if-elses.  There's one below too.  Like this example here shows it's a default import, which lets me match up why it's fine to only have one symbol here.